lib/cylc/cfgspec/globalcfg.py

```diff
@@ -414,15 +414,16 @@ class GlobalConfig(ParsecConfig):
     """
 
     _DEFAULT = None
+    _HOME = os.getenv('HOME') or get_user_home()
     # For working with git clones, just go to minor version number.
     cylc_version = re.sub('-.*', '', CYLC_VERSION)
     CONF_BASE = "global.rc"
     # Site global.rc loc preference: if not in etc/ look in old conf/.
     SITE_CONF_DIR = os.path.join(os.environ["CYLC_DIR"], "etc")
     SITE_CONF_DIR_OLD = os.path.join(os.environ["CYLC_DIR"], "conf")
     # User global.rc loc preference: if not in .cylc/x.y.z/ look in .cylc/.
-    USER_CONF_DIR_1 = os.path.join(os.environ['HOME'], '.cylc', cylc_version)
-    USER_CONF_DIR_2 = os.path.join(os.environ['HOME'], '.cylc')
+    USER_CONF_DIR_1 = os.path.join(_HOME, '.cylc', cylc_version)
```

Comment:
Would it not have been easier to use `$HOME` and replace it only when needed?

Reply:
That's how _HOME is calculated above.